### DISCUSSION TITLE
Document the mapping section model

### DISCUSSION THREAD

**Discussion by martin-braun** [NvChad/NvChad#1622] [Ideas]
Took me a while until I understood that it really makes no difference in what section I put my mappings in my custom `mappings.lua`.

`init.lua` will call `require("core.utils").load_mappings()`, so it will skip picking just a section and load the whole file.

I can see why maintaining sections makes sense for readability, but it introduces a meaning-less structure to the logic (not the user), because I can move my mappings between my sections without results changing at all.

So

```lua
local M = {}

M.disabled = {
    n = {
        ["j"] = "", -- undo navigating through wrapped text, use gj instead
        ["k"] = "", -- undo navigating through wrapped text, use gk instead
    },
    x = {
        ["j"] = "", -- undo navigating through wrapped text, use gj instead
        ["k"] = "", -- undo navigating through wrapped text, use gk instead
        ["p"] = "", -- undo not copying text that got replaced by paste
    },
}

M.general = {
    n = {
        ["<C-n>"] = { "<cmd> enew <CR>", "new buffer" },
        ["<C-f>"] = { "mqgg=G`qzz", "fix indentions"},
        ["<C-s>"] = { "<cmd> w <CR>", "save" },
    },
}

return M
```

<s>is the same as</s>

```lua
local M = {}

M.foobar = {
    n = {
        ["j"] = "", -- undo navigating through wrapped text, use gj instead
        ["k"] = "", -- undo navigating through wrapped text, use gk instead
        ["<C-n>"] = { "<cmd> enew <CR>", "new buffer" },
        ["<C-f>"] = { "mqgg=G`qzz", "fix indentions"},
        ["<C-s>"] = { "<cmd> w <CR>", "save" },
    },
    x = {
        ["j"] = "", -- undo navigating through wrapped text, use gj instead
        ["k"] = "", -- undo navigating through wrapped text, use gk instead
        ["p"] = "", -- undo not copying text that got replaced by paste
    },
}

return M
```

Being new to Neovim, this caused quite a lot of confusion, as I couldn't figure out where those section names are coming from. Only when I was inspecting the `utils.lua` I learned that it really doesn't make a difference.

I do respect the benefit of having more readability with the usage of sections, but I think documentation should clarify this behavior properly.

**Comment by siduck** [comment]
its not the same @martin-braun , for example it'd give out this error 

![image](https://user-images.githubusercontent.com/59060246/194610141-acc387a1-6930-4889-9517-137533223d59.png)

**Reply by martin-braun** [reply]
Isn't this the case when you move the mappings up to have no section at all? That'll be obvious. What wasn't obvious to me was that I can move my mappings between the sections without any side-effects, at least that's what I experienced.

**Reply by siduck** [reply]
without side effects? what do you mean :thinking: . 

Also all mapping tables cant be empty ( " " ) so its obvious that i got that error, only the M.disabled table is ignored

**Reply by martin-braun** [reply]
@siduck Now I understood you. Sure thing, the `disabled` table is special in that regards, I should've tested that. However, I could rename `general` to `foobar` without changing `disabled` and it would still work.

**Reply by siduck** [reply]
@martin-braun I highly doubt. Any keys within tables ( other than disabled table ) will get us some errors

**Reply by martin-braun** [reply]
Well I moved things around without issues and made new a keymap table called `M.coc` for `coc.nvim` related maps. /shrug

**Reply by siduck** [reply]
yes that should work, what I mean was disabling mappings outside the disabled table, oh  nvm that works too :fearful: . Idk why it didnt work before
